Description: Fix dangling pointers
 Resolve FTBFS due to errors about dangling pointers; fixed in upstream ruby
 source code - adapt to truffleruby.
Origin: https://github.com/ruby/ruby/commit/234f4c0bb62b394971eb1044c9c0b1f994d1d4c4
Author: James Page <james.page@chainguard.dev>

diff --git a/src/main/c/prism-gem/src/util/pm_string.c b/src/main/c/prism-gem/src/util/pm_string.c
index 75422fbdf2..cf79885fdd 100644
--- a/src/main/c/prism-gem/src/util/pm_string.c
+++ b/src/main/c/prism-gem/src/util/pm_string.c
@@ -182,7 +182,7 @@ pm_string_mapped_init(pm_string_t *string, const char *filepath) {
 
     if (size == 0) {
         close(fd);
-        const uint8_t source[] = "";
+        static const uint8_t source[] = "";
         *string = (pm_string_t) { .type = PM_STRING_CONSTANT, .source = source, .length = 0 };
         return PM_STRING_INIT_SUCCESS;
     }
@@ -278,7 +278,7 @@ pm_string_file_init(pm_string_t *string, const char *filepath) {
     size_t size = (size_t) sb.st_size;
     if (size == 0) {
         close(fd);
-        const uint8_t source[] = "";
+        static const uint8_t source[] = "";
         *string = (pm_string_t) { .type = PM_STRING_CONSTANT, .source = source, .length = 0 };
         return PM_STRING_INIT_SUCCESS;
     }
diff --git a/src/main/c/yarp/src/util/pm_string.c b/src/main/c/yarp/src/util/pm_string.c
index 75422fbdf2..cf79885fdd 100644
--- a/src/main/c/yarp/src/util/pm_string.c
+++ b/src/main/c/yarp/src/util/pm_string.c
@@ -182,7 +182,7 @@ pm_string_mapped_init(pm_string_t *string, const char *filepath) {
 
     if (size == 0) {
         close(fd);
-        const uint8_t source[] = "";
+        static const uint8_t source[] = "";
         *string = (pm_string_t) { .type = PM_STRING_CONSTANT, .source = source, .length = 0 };
         return PM_STRING_INIT_SUCCESS;
     }
@@ -278,7 +278,7 @@ pm_string_file_init(pm_string_t *string, const char *filepath) {
     size_t size = (size_t) sb.st_size;
     if (size == 0) {
         close(fd);
-        const uint8_t source[] = "";
+        static const uint8_t source[] = "";
         *string = (pm_string_t) { .type = PM_STRING_CONSTANT, .source = source, .length = 0 };
         return PM_STRING_INIT_SUCCESS;
     }
